src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2993,11 +2993,10 @@ ExtractQueryWalker(Node *node, List **queryList)
 		Query *query = (Query *) node;
 
 		(*queryList) = lappend(*queryList, query);
-		walkerResult = query_tree_walker(query, ExtractQueryWalker, queryList,
-										 QTW_EXAMINE_RTES);
+		return query_tree_walker(query, ExtractQueryWalker, queryList, 0);
```

Comment:
We're not dropping that. Instead, if there are no flag (i.e., 0), the walker traverses every part of the input query. The change is required to traverse where clause in addition to the rtable.